File: backend/dtos/src/main/java/io/mateu/remote/dtos/Field.java
package io.mateu.remote.dtos;

import java.util.ArrayList;
import java.util.List;
import lombok.*;

@Data
@Builder
@NoArgsConstructor(access = AccessLevel.PACKAGE)
@AllArgsConstructor(access = AccessLevel.PACKAGE)
public class Field {

  private String id;

  private String type;

  private String stereotype;

  private boolean observed;

  private String caption;

  private String placeholder;

  private String cssClasses;

  private String description;

  private List<Badge> badges;

  private List<Validation> validations = new ArrayList<>();

  private List<Pair> attributes = new ArrayList<>();
}

File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/Field.ts
import Validation from "./Validation";
import Pair from "./Pair";
import Badge from "./Badge";

export default interface Field {

    id: string
    type: string
    stereotype: string
    observed: boolean
    wantsFocus: boolean
    caption: string
    placeholder: string
    cssClasses: string
    description: string
    badges: Badge[]
    validations: Validation[]
    attributes: Pair[]
    colspan: number
    rightAligned: boolean
    bold: boolean

}
